osu.Game/Overlays/DashboardOverlay.cs

```diff
@@ -0,0 +1,179 @@
+                            {
+                                RelativeSizeAxes = Axes.X,
+                                AutoSizeAxes = Axes.Y
+                            }
+                        }
+                    }
+                },
+                loading = new LoadingLayer(content),
+            };
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            background.Colour = ColourProvider.Background5;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            localUser.BindTo(api.LocalUser);
+            localUser.BindValueChanged(_ => onTabChanged());
+            header.Current.BindValueChanged(_ => onTabChanged());
+        }
+
+        protected override void PopIn()
+        {
+            base.PopIn();
+
+            if (!content.Any())
+                header.Current.TriggerChange();
+        }
+
+        protected override void PopOutComplete()
+        {
+            base.PopOutComplete();
+            loadDisplay(null);
+        }
+
+        private void onTabChanged()
+        {
+            lastRequest?.Cancel();
+            cancellationToken?.Cancel();
+
+            loading.Show();
+
+            // We may want to use OnlineViewContainer after https://github.com/ppy/osu/pull/8044 merge
+            if (!api.IsLoggedIn)
+            {
+                loadDisplay(null);
+                return;
+            }
+
+            var request = createScopedRequest();
+            lastRequest = request;
+
+            if (request == null)
+            {
+                loadDisplay(null);
+                return;
+            }
+
+            request.Success += () => Schedule(() => loadDisplay(createDisplayFromResponse(request)));
+            request.Failure += _ => Schedule(() => loadDisplay(null));
+
+            api.Queue(request);
+        }
+
+        private void loadDisplay(Drawable display)
+        {
+            scrollFlow.ScrollToStart();
+
+            if (display == null)
+            {
+                content.Clear();
+                loading.Hide();
+                return;
+            }
+
+            LoadComponentAsync(display, loaded =>
+            {
+                loading.Hide();
+                content.Child = loaded;
+            }, (cancellationToken = new CancellationTokenSource()).Token);
+        }
+
+        private Drawable createDisplayFromResponse(APIRequest request)
+        {
+            switch (request)
+            {
+                case GetFriendsRequest friendsRequest:
+                    return new FriendDisplay(friendsRequest.Result);
+            }
+
+            return null;
+        }
+
+        private APIRequest createScopedRequest()
+        {
+            switch (header.Current.Value)
+            {
+                case HomeOverlayTabs.Friends:
+                    return new GetFriendsRequest();
+            }
+
+            return null;
```

Comment:
This one I'd say should throw an `ArgumentOutOfRangeException` as it's clear that if the enum is missing, something is just not implemented (don't think any of the future tabs in this overlay will be fully offline).